src/app/view-config-utils.js

```diff
@@ -162,6 +173,31 @@ function initializeAuto(config) {
  * @param {object} config The view config prop.
  */
 export function initialize(config) {
+  // Add a log message when there are additionalProperties in the coordination space that
+  // do not appear in the view config JSON schema,
+  // with a note that this indicates either a mistake or custom coordination type usage.
+  const coordinationTypesInConfig = Object.keys(config.coordinationSpace || {});
+  const allCoordinationTypes = getCoordinationTypes();
+  const unknownCoordinationTypes = difference(coordinationTypesInConfig, allCoordinationTypes);
+  if (unknownCoordinationTypes.length > 0) {
+    throw new Error(`The following coordination types are not recognized: [${unknownCoordinationTypes}].\nIf these are plugin coordination types, ensure that they have been properly registered.`);
+  }
+  // Add a log message when there are views in the layout that are neither
+  // core views nor registered plugin views.
+  const viewTypesInConfig = config.layout.map(c => c.component);
+  const allViewTypes = getViewTypes();
+  const unknownViewTypes = difference(viewTypesInConfig, allViewTypes);
+  if (unknownViewTypes.length > 0) {
+    throw new Error(`The following view types are not recognized: [${unknownViewTypes}].\nIf these are plugin view types, ensure that they have been properly registered.`);
+  }
+  // Add a log message when there are file definitions with neither
+  // core nor registered plugin file types.
+  const fileTypesInConfig = config.datasets.flatMap(d => d.files.map(f => f.fileType));
+  const allFileTypes = getFileTypes();
+  const unknownFileTypes = difference(fileTypesInConfig, allFileTypes);
+  if (unknownFileTypes.length > 0) {
+    throw new Error(`The following file types are not recognized: [${unknownFileTypes}].\nIf these are plugin file types, ensure that they have been properly registered.`);
+  }
```

Comment:
Maybe wrap this in a different function so we can just call it like `validatePlugInTypes(config)`

Reply:
Done in https://github.com/vitessce/vitessce/pull/1212/commits/8964b41c74fd7782e5c7f7ba901ab8a2966b846b#diff-b4ecb97b806489d9cb877f3bd3d271fe1ea45dc4fc3e80705e625b992e1206b0L165